tests/unit/test_messageDialog.py

```diff
@@ -0,0 +1,903 @@
+		self.dialog.addApplyButton().addCloseButton()
+		self.dialog.setFallbackAction(ReturnCode.CLOSE)
+		command = self.dialog._getFallbackActionOrFallback()
+		self.assertEqual(command.returnCode, ReturnCode.CLOSE)
+		self.assertIsNotNone(command)
+		self.assertTrue(command.closesDialog)
+
+	def test_getFallbackActionOrFallbackEscapeIdNotACommand(self):
+		"""Test that calling _getFallbackActionOrFallback on a dialog whose EscapeId is not a command falls back to returning the default focus."""
+		self.dialog.addOkCancelButtons()
+		super(MessageDialog, self.dialog).SetEscapeId(ReturnCode.CLOSE)
+		command = self.dialog._getFallbackActionOrFallback()
+		self.assertEqual(command.returnCode, ReturnCode.OK)
+		self.assertIsNotNone(command)
+		self.assertTrue(command.closesDialog)
+
+	def test_getFallbackActionEscapeCode_None(self):
+		"""Test that setting EscapeCode to None causes _getFallbackAction to return None."""
+		self.dialog.addOkCancelButtons()
+		self.dialog.SetEscapeId(EscapeCode.NO_FALLBACK)
+		self.assertIsNone(self.dialog._getFallbackAction())
+
+
+class Test_MessageDialog_Threading(WxTestBase):
+	def test_newOnNonmain(self):
+		"""Test that creating a MessageDialog on a non GUI thread fails."""
+		with ThreadPoolExecutor(max_workers=1) as tpe:
+			with self.assertRaises(RuntimeError):
+				tpe.submit(MessageDialog.__new__, MessageDialog).result()
+
+	def test_initOnNonMain(self):
+		"""Test that initializing a MessageDialog on a non-GUI thread fails."""
+		dlg = MessageDialog.__new__(MessageDialog)
+		with ThreadPoolExecutor(max_workers=1) as tpe:
+			with self.assertRaises(RuntimeError):
+				tpe.submit(dlg.__init__, None, "Test").result()
+
+	def test_showOnNonMain(self):
+		# self.app = wx.App()
+		"""Test that showing a MessageDialog on a non-GUI thread fails."""
+		dlg = MessageDialog(None, "Test")
+		with ThreadPoolExecutor(max_workers=1) as tpe:
+			with self.assertRaises(RuntimeError):
+				tpe.submit(dlg.Show).result()
+
+	def test_showModalOnNonMain(self):
+		"""Test that showing a MessageDialog modally on a non-GUI thread fails."""
+		# self.app = wx.App()
+		dlg = MessageDialog(None, "Test")
+		with ThreadPoolExecutor(max_workers=1) as tpe:
+			with self.assertRaises(RuntimeError):
+				tpe.submit(dlg.ShowModal).result()
+
+
+@patch.object(wx.Dialog, "Show")
+class Test_MessageDialog_Show(MDTestBase):
+	def test_showNoButtons(self, mocked_show: MagicMock):
+		"""Test that showing a MessageDialog with no buttons fails."""
+		with self.assertRaises(RuntimeError):
+			self.dialog.Show()
+		mocked_show.assert_not_called()
+
+	def test_show(self, mocked_show: MagicMock):
+		"""Test that showing a MessageDialog works as expected."""
+		self.dialog.addOkButton()
+		self.dialog.Show()
+		mocked_show.assert_called_once()
+
+
+@patch("gui.mainFrame")
+@patch.object(wx.Dialog, "ShowModal")
+class Test_MessageDialog_ShowModal(MDTestBase):
+	def test_showModalNoButtons(self, mocked_showModal: MagicMock, _):
```

Comment:
could you clarify what the underscore mean in this context?

Reply:
It's just a conventional way of showing that we don't care about the mock object that is patching `mainFrame`.

Reply:
Would you like me to add a comment explaining that?

Reply:
all good - just wanted to learn. It's somewhat self-explanatory

Reply:
I know that we are only in unit tests and that using `_` this way is a common practice.
Though, in projects where `_` is already used by `gettext`, it's not recommended to also use `_` for unused variables to avoid confusion.

That's really minor though, so feel free to ignore this comment if you wish.